ext/compressors/iaa/iaacodec/include/IAACompressionCodecDeflate.h

```diff
@@ -0,0 +1,218 @@
+#pragma once
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <unistd.h>
+#include <chrono>
+#include <sys/mman.h>
+#include <iostream>
+#include <atomic>
+#include <qpl/qpl.h>
+#include <vector>
+#include <map>
+#include <x86intrin.h>
+#include <memory>
+#include <thread>
+#include <wiredtiger.h>
+#include <wiredtiger_ext.h>
+namespace DB
+{
+
+class DeflateJobHWPool
+{
+public:
+    DeflateJobHWPool(WT_COMPRESSOR *compressor, WT_SESSION *session);
+    ~DeflateJobHWPool();
+    static DeflateJobHWPool & instance(WT_COMPRESSOR *compressor = nullptr, WT_SESSION *session = nullptr);
+    static constexpr auto jobPoolSize = 256;
+    static constexpr qpl_path_t PATH = qpl_path_hardware;
+    static qpl_job * jobPool[jobPoolSize];
+    static std::atomic_bool jobLock[jobPoolSize];
+    bool jobPoolEnabled;
+    std::atomic_uint64_t jobIndex;
+
+    inline bool jobPoolReady()
+    {
+        return jobPoolEnabled;
+    }
+    inline qpl_job * acquireJob(uint32_t * job_id)
+    {
+        if(jobPoolEnabled)
+        {
+            uint32_t retry = 0;
+            auto index = jobIndex.fetch_add(1) % jobPoolSize;
+            while (tryLockJob(index) == false)
+            {
+                retry++;
+                if (retry > jobPoolSize)
+                {
+                    return nullptr;
+                }
+                index = jobIndex.fetch_add(1) % jobPoolSize;
+            }
+            *job_id = jobPoolSize - index;
+            return jobPool[index];
+        }
+        else
```

Comment:
For one-line statements, we usually don't add brackets:
```suggestion
        else
            return nullptr;
```
Please update this throughout your changes.

Reply:
Good!
We will remove brackets, Thanks!